.changeset/new-frogs-punch.md

```diff
@@ -0,0 +1,7 @@
+---
+'@backstage/plugin-kubernetes-backend': minor
+---
+
+Integrators can now bring their own auth strategies through the use of the `addAuthStrategy` method on `KubernetesBuilder`.
+
+**BREAKING** on the slight chance you were using the `setAuthTranslatorMap` method on `KubernetesBuilder`, it has been removed along with the entire `KubernetesAuthTranslator` interface. This notion has been replaced with the more focused concept of an `AuthenticationStrategy`. Converting a translator to a strategy should not be especially difficult.
```

Comment:
I thought this was rare, but just learned [here](https://github.com/backstage/backstage/issues/13793#issuecomment-1719616679) that @gavlyukovskiy is using this method.

Reply:
Oh, thank you for tagging me! We merely use it as a hacky workaround and it is indeed doesn't look difficult to change existing translator to a new interface, I'll keep this in mind :+1:

Reply:
Though, if I may - it would be quite nice if we could have a label for setting `x-k8s-aws-id` (`clusterDetails.name`), that way we wouldn't have to have our implementation
```
      token: await this.getBearerToken(
        clusterDetails.name,
        clusterDetails.authMetadata[ANNOTATION_KUBERNETES_AWS_ASSUME_ROLE],
        clusterDetails.authMetadata[ANNOTATION_KUBERNETES_AWS_EXTERNAL_ID],
      ),
    };
```

Reply:
I agree that the `x-k8s-aws-id` parameter used by `AwsIamStrategy` should ideally be sourced from a distinct authMetadata field and not the `name` field from the cluster, but personally I'd like to handle _that_ change separately. I think there was a prior closed PR that created a new field on ClusterDetails which could override the `name` field for exactly your use-case.